### ISSUE TITLE
Add configurable limit on the number of parallel streams that can be active/instantiated at one time

### ISSUE THREAD

**Issue by korydraughn** [irods/irods_client_http_api#79]
The HTTP API currently allows an infinite number of parallel streams to be opened. This is really bad for systems which cannot afford to go down often.

To help with this situation, the HTTP API needs to provide a config option to limit the number of active parallel streams. This protects the HTTP API and the iRODS server.

If a client requests _N_ streams and instantiating them results in exceeding the limit, the HTTP API needs to return an appropriate HTTP status code to inform the client that they should try again later due to resource limitations.

**Comment by trel** [comment]
This is independent of the current settings for number of threads incoming, and size of connection pool to iRODS...

So this is a total, across all threads/connections.   Will have to do some min/max comparisons on the configs to make sure they make sense together.

**Comment by korydraughn** [comment]
Correct. This has more to do with how parallel writes are supported.

When a client wants to do a parallel write, they have to tell the HTTP API how many streams they want. We already have an option for that limiting that. What we don't have is an option to limit the total number of streams across all clients.

Initiating parallel writes causes the HTTP API to create dedicated dstream objects for a client. They are not shared and they persist across requests.

**Comment by trel** [comment]
But we already have 

```
        "requests": {
            // The number of threads dedicated to servicing client requests.
            // When adjusting this value, consider adjusting "background_io/threads"
            // and "irods_client/connection_pool/size" as well.
            "threads": 3,
```

Does that not limit the total itself?

**Comment by korydraughn** [comment]
It doesn't.

That only limits the number of requests that can be **accepted** asynchronously.

The HTTP API can create any number of dstream objects. It does not consider anything other than the parallel write stream count limit when dealing with parallel writes.

**Comment by trel** [comment]
Okay, so we'll have 3-4-5 different limits that will interact with each other, kind of...

**Comment by korydraughn** [comment]
Correct.
